lib/Sema/TypeCheckType.cpp

```diff
@@ -4978,13 +4978,38 @@ TypeResolver::resolveExistentialType(ExistentialTypeRepr *repr,
 
 NeverNullType TypeResolver::resolveMetatypeType(MetatypeTypeRepr *repr,
                                                 TypeResolutionOptions options) {
+  auto &ctx = getASTContext();
+  auto baseOptions = options.withContext(TypeResolverContext::MetatypeBase);
+
   // The instance type of a metatype is always abstract, not SIL-lowered.
   if (options.is(TypeResolverContext::ExistentialConstraint))
-    options |= TypeResolutionFlags::DisallowOpaqueTypes;
-  options = options.withContext(TypeResolverContext::MetatypeBase);
-  auto ty = resolveType(repr->getBase(), options);
+    baseOptions |= TypeResolutionFlags::DisallowOpaqueTypes;
+
+  // The parser assumes 'X.Type' is a metatype representation, but it might be
```

Comment:
Can we get rid of metatype representations and model them as a MemberTypeRepr?

Reply:
🙂

Reply:
Oh, but we would need two bits to disambiguate between `X.Type`/`X.Protocol` and ``` X.`Type` ```/``` X.`Protocol` ```. Does that sound alright?

Reply:
Or just one `isEscaped` bit for `SimpleIdentTypeRepr` really.

Reply:
We absolutely need that bit to disambiguate in expression context anyway. Currently, both of these type-check to metatypes:

```swift
extension Int {
  struct `Type` {}
}

_ = Int.`Type`.self
_ = Int.Type.self
```

Reply:
Let’s start with #70467.